pkg/controller/controlplane/genericactuator/actuator.go

```diff
@@ -165,6 +165,9 @@ func (a *actuator) Reconcile(
 	if cp.Spec.Purpose != nil && *cp.Spec.Purpose == extensionsv1alpha1.Exposure {
 		return a.reconcileControlPlaneExposure(ctx, cp, cluster)
 	}
+	if cp.Spec.ProviderConfig == nil {
```

Comment:
We should not enforce this in the generic actuator. It's a provider-specific decision whether this is needed or not.

Reply:
any follow up @danielfoehrKn ?